server/stream.go

```diff
@@ -1412,6 +1412,13 @@ func (mset *stream) updateWithAdvisory(config *StreamConfig, sendAdvisory bool)
 					mset.sources[s.iname] = si
 					mset.setStartingSequenceForSource(s.iname)
 					mset.setSourceConsumer(s.iname, si.sseq+1)
+				} else if oFilter != s.FilterSubject {
+					if si, ok := mset.sources[s.iname]; ok {
+						mset.cancelSourceInfo(si)
```

Comment:
I believe this is not necessary since setSourceConsumer will cancel if 'iname' is found in mset.sources, but that is ok I guess.

Reply:
yes, agreed.